packages/redux/src/exchanges/reducer.ts

```diff
@@ -0,0 +1,128 @@
+    case actionTypes.FETCH_EXCHANGE_FAILURE:
+      return action.payload.error;
+    case actionTypes.CREATE_EXCHANGES_REQUEST:
+    case actionTypes.CREATE_EXCHANGE_BOOK_REQUESTS_REQUEST:
+    case actionTypes.CREATE_EXCHANGE_FILTERS_REQUEST:
+    case actionTypes.FETCH_EXCHANGE_BOOK_REQUEST_REQUEST:
+    case actionTypes.FETCH_EXCHANGE_REQUEST:
+      return INITIAL_STATE.error;
+    default:
+      return state;
+  }
+};
+
+const isLoading = (state = INITIAL_STATE.isLoading, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.CREATE_EXCHANGES_REQUEST:
+    case actionTypes.CREATE_EXCHANGE_BOOK_REQUESTS_REQUEST:
+    case actionTypes.CREATE_EXCHANGE_FILTERS_REQUEST:
+    case actionTypes.FETCH_EXCHANGE_BOOK_REQUEST_REQUEST:
+    case actionTypes.FETCH_EXCHANGE_REQUEST:
+      return true;
+    case actionTypes.CREATE_EXCHANGES_SUCCESS:
+    case actionTypes.CREATE_EXCHANGES_FAILURE:
+    case actionTypes.CREATE_EXCHANGE_BOOK_REQUESTS_SUCCESS:
+    case actionTypes.CREATE_EXCHANGE_BOOK_REQUESTS_FAILURE:
+    case actionTypes.CREATE_EXCHANGE_FILTERS_SUCCESS:
+    case actionTypes.CREATE_EXCHANGE_FILTERS_FAILURE:
+    case actionTypes.FETCH_EXCHANGE_BOOK_REQUEST_SUCCESS:
+    case actionTypes.FETCH_EXCHANGE_BOOK_REQUEST_FAILURE:
+    case actionTypes.FETCH_EXCHANGE_SUCCESS:
+    case actionTypes.FETCH_EXCHANGE_FAILURE:
+      return INITIAL_STATE.isLoading;
+    default:
+      return state;
+  }
+};
+
+const result = (state = INITIAL_STATE.result, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_EXCHANGE_SUCCESS:
+      return action.payload.result;
+    default:
+      return state;
+  }
+};
+
+export const exchangeFilters = createReducerWithResult(
+  'CREATE_EXCHANGE_FILTERS',
+  INITIAL_STATE.exchangeFilters,
+  actionTypes,
+  false,
+  false,
+  actionTypes.RESET_EXCHANGE_FILTERS_STATE,
+);
+
+export const exchangeBookRequest = createReducerWithResult(
+  'FETCH_EXCHANGE_BOOK_REQUEST',
+  INITIAL_STATE.exchangeBookRequest,
+  actionTypes,
+  false,
+  false,
+  actionTypes.RESET_EXCHANGE_BOOK_REQUEST_STATE,
+);
+
+export const getError = (state: T.ExchangesState) => state.error;
+export const getIsLoading = (state: T.ExchangesState) => state.isLoading;
+export const getResult = (state: T.ExchangesState) => state.result;
+export const getExchangeFilters = (state: T.ExchangesState) =>
+  state.exchangeFilters;
+export const getExchangeBookRequest = (state: T.ExchangesState) =>
+  state.exchangeBookRequest;
+
+/**
+ * Reducer for exchanges state.
+ *
+ * @param state  - Current redux state.
+ * @param action - Action dispatched.
+ *
+ * @returns New state.
+ */
+const reducer: Reducer<T.ExchangesState> = combineReducers({
+  error,
+  isLoading,
+  result,
+  exchangeFilters,
+  exchangeBookRequest,
+});
+
+const exchangesReducer: Reducer<T.ExchangesState> = (state, action) => {
+  if (
+    action.type === LOGOUT_SUCCESS ||
+    action.type === actionTypes.RESET_EXCHANGES
```

Comment:
The reset exchanges action type is being handled but there is not an action being exported that the consumer can use to dispatch this action type. The same for the RESET_EXCHANGE_BOOK_REQUEST_STATE and RESET_EXCHANGE_FILTERS_STATE action types.

Reply:
Done, created the missing actions.